@here/harp-mapview/lib/SphereHorizon.ts

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EarthConstants, MathUtils, Projection, ProjectionType } from "@here/harp-geoutils";
+import { assert, Math2D } from "@here/harp-utils";
+import * as THREE from "three";
+
+import { MapViewUtils } from "./Utils";
+
+const twoPi = Math.PI * 2;
+
+// Keep counter clockwise order.
+export enum CanvasSide {
+    Bottom,
+    Right,
+    Top,
+    Left
+}
+
+/**
+ * Class computing horizon tangent points and intersections with canvas for spherical projection.
+ *
+ * @remarks
+ *
+ * The horizon for a sphere is a circle on a plane with a normal matching the sphere normal at the
```

Comment:
It might be me, but I have read this now a couple of times, and it still confuses me.

So we have a circle on a plane as the horizon, I get that..
Now with the normal and center point I am not sure I understand: 

The normal of the plane is then a vector from the sphere center to the camera position??
The Plane always contains the sphereCenter, right? 
So the Circle Center Point would be the same as the Sphere Center Point?
The circle then would be the intersection of the plane with the sphere?

Maybe be a bit more verbose, and I have to agree with Nino, maybe it´s due to the formatting in the github diff, but the ascii does not help.

Reply:
sorry, I was still working on the documentation, I'll upload more informative diagrams today. 

But answering your question: the horizon circle does not have generally the center on the sphere center. The horizon center is always at the line joining the earth center and the camera. The further away the camera is, the closer  the horizon center to the earth center becomes. When the camera is at infinity, the horizon center is at the earth center and therefore it becomes a great circle.

Reply:
I've rewritten the remarks and added several diagrams.

Reply:
thanks, for the clarification. Make sense to me now.